File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/domain/repository/TwitterRepository.scala
package com.github.kazuhito_m.twitterbattler.primitive.domain.repository

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.social.twitter.api.{Twitter, TwitterProfile}
import org.springframework.stereotype.Repository

@Repository
class TwitterRepository {

  @Autowired
  val twitter: Twitter = null

  /**
    * Twitterから「指定されたIDのプロフィール情報」を取得し、オブジェクトで返す。
    *
    * @param id TwitterID。
    * @return プロフィールオブジェクト。
    */
  def getProfile(id: String): TwitterProfile = twitter.userOperations().getUserProfile(id)

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/application/repository/TwitterRepository.java
package com.github.kazuhito_m.twitterbattler.primitive.application.repository;

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battler.BattlerIdentifier;
import com.github.kazuhito_m.twitterbattler.primitive.domain.model.twitter.TwitterUser;

import java.util.List;
import java.util.Set;

public interface TwitterRepository {
    BattlerIdentifier convertScreenNameToId(String screenName);

    List<TwitterUser> getRandomAccounts();

    Set<TwitterUser> searchAccountsFromTweetWord(String word);

    TwitterUser getProfile(BattlerIdentifier identifier);

    List<Long> getFollowers(BattlerIdentifier identifier);

    List<Long> getFollows(BattlerIdentifier identifier);
}
